fpmsyncd/routesync.cpp

```diff
@@ -1508,3 +1735,143 @@ void RouteSync::onWarmStartEnd(DBConnector& applStateDb)
         SWSS_LOG_NOTICE("Warm-Restart reconciliation processed.");
     }
 }
+
+/*
+ * Get nexthop group key as string
+ * @arg id     next hop group id
+ *
+ * Return nexthop group key
+ */
+#ifdef HAVE_NEXTHOP_GROUP
+const string RouteSync::getNextHopGroupKeyAsString(uint32_t id) const
+{
+    return string("ID") + to_string(id);
+}
+
+/*
+ * update the nexthop group entry
+ * @arg nh_id     nexthop group id
+ *
+ */
+void RouteSync::updateNextHopGroup(uint32_t nh_id)
+{
+    auto git = m_nh_groups.find(nh_id);
+    if(git == m_nh_groups.end())
+    {
+        SWSS_LOG_INFO("Nexthop not found: %d", nh_id);
+        return;
+    }
+
+    NextHopGroup& nhg = git->second;
+
+    if(nhg.installed)
+    {
+        //Nexthop group already installed
+        return;
+    }
+    nhg.installed = true;
+    updateNextHopGroupDb(nhg);
+}
+
+/*
+ * delete the nexthop group entry
+ * @arg nh_id     nexthop group id
+ *
+ */
+void RouteSync::deleteNextHopGroup(uint32_t nh_id)
+{
+    auto git = m_nh_groups.find(nh_id);
+    if(git == m_nh_groups.end())
+    {
+        SWSS_LOG_INFO("Nexthop not found: %d", nh_id);
```

Comment:
Shouldn't this be a warn or error message?

Reply:
I have corrected it to SWSS_LOG_ERROR.